connmgr.go

```diff
@@ -51,6 +51,15 @@ func (d *Daemon) doConnManager(req *pb.Request) *pb.Response {
 		d.host.ConnManager().TrimOpenConns(ctx)
 		return okResponse()
 
+	case pb.ConnManagerRequest_DISCONNECT_PEER:
+		p, err := peer.IDFromBytes(req.ConnManager.GetPeer())
+		if err != nil {
+			return errorResponse(err)
+		}
+
+		d.host.Network().ClosePeer(p)
```

Comment:
ClosePeer returns an `error`. It's a synchronous call as it is, so we're going to wait until connections are closed anyway. Can we propagate the error to the caller?

Reply:
Sure, and we should.